File: frontend/src/types/update.ts
export interface UpdateCheckResult {
  available: boolean;
  enabled: boolean;
  currentVersion: string;
  latestVersion: string;
  releaseName: string;
  notes: string;
  releaseURL: string;
}

export function formatError(err: unknown): string {
  if (err instanceof Error) return err.message;
  if (typeof err === "string") return err;
  return "操作失败";
}

File: backend/model/update.go
package model

type UpdateCheckResult struct {
	Available      bool   `json:"available"`
	Enabled        bool   `json:"enabled"`
	CurrentVersion string `json:"currentVersion"`
	LatestVersion  string `json:"latestVersion"`
	ReleaseName    string `json:"releaseName"`
	Notes          string `json:"notes"`
	ReleaseURL     string `json:"releaseURL"`
}
